app/src/main/java/com/hover/stax/balances/BalancesFragment.java

```diff
@@ -95,25 +99,24 @@ private void updateServices(List<Channel> channels, View view) {
 		recyclerView.setVisibility(channels != null && channels.size() > 0 ? VISIBLE : GONE);
 
 		((StaxCardView) view.findViewById(R.id.balance_card)).backButton.setVisibility(channels != null && channels.size() > 0  ? VISIBLE : GONE);
-		if(channels != null && channels.size() > 0) {
-			channelDropdown.toggleLink(true);
-			channelDropdown.removeObservers(channelDropdownViewModel, getViewLifecycleOwner());
-		}else {
-			channelDropdown.toggleLink(false);
-			channelDropdown.setObservers(channelDropdownViewModel, channelDropdown, getViewLifecycleOwner());
-		}
-	}
 
-	private void refreshBalances(View v) {
-		if (channelDropdown.getHighlighted() != null
-					|| (balancesViewModel.getActions().getValue() !=null
-					&& balancesViewModel.getActions().getValue().size() !=0)) {
+		toggleLink(channels != null && channels.size() > 0);
+		channelDropdown.setObservers(channelDropdownViewModel, getActivity());
+	}
 
-					balancesViewModel.setAllRunning(v.getContext());
+	public void toggleLink(boolean show) {
```

Comment:
Moved the link out of the ChannelDropdown since it is only used and one place and the dropdown no longer replaces it in the design.